ppmclibs/tinycv_impl.cc

```diff
@@ -645,6 +667,47 @@ void image_map_raw_data_rgb555(Image* a, const unsigned char* data)
     }
 }
 
+void image_map_raw_data_uyvy(Image* a, const unsigned char* data)
+{
+    for (int y = 0; y < a->img.rows; y++) {
+        for (int x = 0; x < a->img.cols; x += 2) {
+            int u  = data[(y * a->img.cols + x) * 2 + 0];
+            int y1 = data[(y * a->img.cols + x) * 2 + 1];
+            int v  = data[(y * a->img.cols + x) * 2 + 2];
+            int y2 = data[(y * a->img.cols + x) * 2 + 3];
```

Comment:
It would probably be better to avoid repeating sub-expressions like that:
```suggestion
            int offset = (y * a->img.cols + x) * 2;
            int u  = data[offset + 0];
            int y1 = data[offset + 1];
            int v  = data[offset + 2];
            int y2 = data[offset + 3];
```
(Maybe you can come up with a better variable name than me.)